ambry-api/src/test/java/com.github.ambry/account/AccountContainerTest.java

```diff
@@ -269,49 +277,58 @@ public void testAccountBuilder() throws JSONException {
    */
   @Test
   public void testContainerBuilder() throws JSONException {
-    ContainerBuilder containerBuilder;
     for (int i = 0; i < CONTAINER_COUNT; i++) {
       // build a container with arguments supplied
-      containerBuilder =
+      ContainerBuilder containerBuilder =
           new ContainerBuilder(refContainerIds.get(i), refContainerNames.get(i), refContainerStatuses.get(i),
-              refContainerDescriptions.get(i), refContainerPrivacyValues.get(i), refAccountId);
+              refContainerDescriptions.get(i), refContainerEncryptionValues.get(i),
+              refContainerPreviousEncryptionValues.get(i), refContainerCachingValues.get(i),
+              refContainerMediaScanDisabledValues.get(i), refAccountId);
       Container containerFromBuilder = containerBuilder.build();
-      assertContainer(containerFromBuilder, i);
+      assertContainer(containerFromBuilder, i, LATEST_CONTAINER_JSON_VERSION);
```

Comment:
I am not very comfortable in the way we go about constructing container w/ new fields and once it is serialized and deserialized, it resets some of the fields to defaults. Can we set the defaults in the constructor itself. Sorry to drag this patch more. But I feel like most of the tests for now tests V2 container and not V1.

Reply:
I have made this change and changed the test setup to parameterize the serialization version